models/repo/user_repo.go

```diff
@@ -177,3 +177,22 @@ func GetIssuePostersWithSearch(ctx context.Context, repo *Repository, isPull boo
 		Limit(30).
 		Find(&users)
 }
+
+// GetIssuePostersWithSearchWithoutRepo clone of GetIssuePostersWithSearch without repo
+func GetIssuePostersWithSearchWithoutRepo(ctx context.Context, search string, isShowFullName bool) ([]*user_model.User, error) {
```

Comment:
Thanks for the PR. I'm concerned with the (near) duplication of these functions, as it does introduce the possibility of drift during refactors where only one of the two functions is changed.
Could something like the following work?
```
func WithRepo(...) (...) {
repoCond = ...
ret shared(repoCond)
}
func WithoutRepo(...) (...) {
repoCond = // 1==1  or some other noop condition
ret shared(repoCond)
}
func shared(...) (...) //...
```

Please let me know if that's unclear, and I can give a more detailed example than the one above.

Reply:
since there was two duplicate functions (issuePosters, and GetIssuePostersWithSearch both with/without repo ), i had to do what you suggested twice, but i found a way to do it without duplication, let me know if thats correct @techknowlogick